### PR TITLE
API Group

### PR DESCRIPTION
**Pull request by huynhlinh2904**

### CODE REVIEW

File: src/apis/conversations/conversations.controller.ts [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
+  @Put()
+  async updateConversationName(@Param('id') id: string, @Body() dto: ConversationDTO) {
+    return await this.conversationService.updateConversation(id, dto);
+  }
+
+  @Public()
+  @Delete()
+  async deleteConversationName(@Param('id') id: string) {
+    return await this.conversationService.deleteConversation(id);
+  }
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.controller.ts]: why using public here

File: src/apis/conversations/conversations.service.ts [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
+    return await this.conversationRepository.save(conversation);
+  }
+
+  async deleteConversation(id: string) {
+    const conversation = await this.conversationRepository.findOne({ where: { id } });
+    if (!conversation) {
+      throw new HttpException('not found conversation', HttpStatus.NOT_FOUND);
+    }
+    conversation.isActive = false;
+    //conversation.deletedAt = new Date();
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.service.ts]: remove this

**Review by baosphoa0403 — CHANGES_REQUESTED** [review]

File: src/apis/conversations/conversations.repository.ts [hunk 3]
```diff
@@ -1,13 +1,43 @@
...
+      query.where('conversations.name LIKE :name', { name: `%${name}%` });
+    }
+    query.andWhere('conversations.isActive IN (true, false)', { isActive: [true, false] });
+    return query.getMany();
+  }
+
+  async findAll() {
+    const query = await this.generalQueryBuilder();
+    query.leftJoinAndSelect('conversations.userConversationEntities', 'uc')
+    query.leftJoinAndSelect('uc.userEntity', 'u')
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.repository.ts]: query.leftJoinAndSelect('conversations.userConversationEntities', 'uc').leftJoinAndSelect('uc.userEntity', 'u')

File: src/apis/conversations/conversations.repository.ts [hunk 4]
```diff
@@ -1,13 +1,43 @@
...
+    return query.getMany();
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.repository.ts:32]: lack of await

File: src/apis/conversations/conversations.repository.ts [hunk 5]
```diff
@@ -1,13 +1,43 @@
...
+  generalQueryBuilder(): SelectQueryBuilder<ConversationEntity> {
+    return this.ConversationRepository.createQueryBuilder('conversations');
+  }
+
+  async findConversationByName(name: string): Promise<ConversationEntity[]> {
+    const query = this.generalQueryBuilder();
+    if (name) {
+      query.where('conversations.name LIKE :name', { name: `%${name}%` });
+    }
+    query.andWhere('conversations.isActive IN (true, false)', { isActive: [true, false] });
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.repository.ts]: explain for me write code here

File: src/apis/conversations/conversations.module.ts [hunk 6]
```diff
@@ -4,11 +4,13 @@ import { ConversationsService } from './conversations.service';
 import { TypeOrmModule } from '@nestjs/typeorm';
 import { ConversationEntity } from './entity/conversations.entity';
 import { ConversationRepository } from './conversations.repository';
+import { UserEntity } from '../user/entity/user.entity';
+import { UserConversationEntity } from '../user-conversations/entity/user-conversations.entity';
 
 @Module({
-  imports: [TypeOrmModule.forFeature([ConversationEntity])],
+  imports: [TypeOrmModule.forFeature([ConversationEntity, UserEntity, UserConversationEntity])],
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.module.ts:11]: Error Must be inject UserService An UserConversationService here

File: src/apis/conversations/conversations.controller.ts [hunk 7]
```diff
@@ -1,7 +1,51 @@
...
+  @Get()
+  async findAllConversation() {
+    return await this.conversationService.findAll();
+  }
+
+
+  @Get(':userId')
+  async findUserByUserId(@Param('userId') userId: string): Promise<ConversationEntity> {
+    return await this.conversationService.findUserByUserId(userId);
+  }
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.controller.ts:28]: why need api :userId

File: src/apis/conversations/conversations.controller.ts [hunk 8]
```diff
@@ -1,7 +1,51 @@
...
+  @Get('/search')
+  async searchByName(@Query('name') search: string): Promise<ConversationEntity[]> {
+    return await this.conversationService.findConversationsByName(search);
+  }
+
+
+  @Get()
+  async findAllConversation() {
+    return await this.conversationService.findAll();
+  }
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.controller.ts]: api @Get('/search') same  @Get()

File: env.prod [hunk 9]
```diff
@@ -0,0 +1,10 @@
+APP_PORT=8080
+NODE_ENV=production
+POSTGRES_HOST=localhost
```
Comment (baosphoa0403) [inline env.prod]: remove file

File: docker-app-chat/docker-compose.yml [hunk 10]
```diff
@@ -1,45 +1,45 @@
 version: '3.5'
```
Comment (baosphoa0403) [inline docker-app-chat/docker-compose.yml]: please revert file

File: src/apis/conversations/conversations.service.ts [hunk 11]
```diff
@@ -1,7 +1,77 @@
...
+      where: { userId: data.userId, conversationID: data.conversationId }
+    });
+
+    if (!user) {
+      throw new HttpException('User Not Found', HttpStatus.NOT_FOUND);
+    }
+
+    if (user.isHost === false) {
+      throw new HttpException('User Unauthorized to update this conversation', HttpStatus.FORBIDDEN);
+    }
```
Comment (baosphoa0403) [inline src/apis/conversations/conversations.service.ts]: Must be use guard

**Review by baosphoa0403 — CHANGES_REQUESTED** [review]

File: src/apis/conversations/conversations.repository.ts (hunk 5 above)
Comment (huynhlinh2904) [reply src/apis/conversations/conversations.repository.ts]: this code get conversation with isActive True and False(if hasn't  andWhere is get with isActive True )

File: src/apis/conversations/conversations.controller.ts (hunk 7 above)
Comment (huynhlinh2904) [reply src/apis/conversations/conversations.controller.ts:28]: I think user login in system will get conversation for that user

**Pull request merged**